### PR TITLE
Solution assignment-5

### PR DESCRIPTION
**Pull request by ndomanova**

### CODE REVIEW

File: group/assignment-5/Node.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    public HashMap<Character, Node> getChildren() {
+        return children;
+    }
+
+    public Node getChild(char c) {
+        if (children.containsKey(c)) {
+            return children.get(c);
+        } else {
+            Node node = new Node(children.size());
```
Comment (angelina-nedelcu) [inline group/assignment-5/Node.java]: This function should only do one thing: get child. Someone that just sees the name of the method without implementation, will expect null to be returned if the char c is not in the children list and his/her last intention would be to create another node.

You can create another function createNode / insertNode where to put the else branch and use getChild to verify if it is already there.

File: group/assignment-5/Node.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class Node {
+    private HashMap<Character, Node> children;
```
Comment (angelina-nedelcu) [inline group/assignment-5/Node.java]: Isn't it a bit strange that the Node does not know its own Character value?

File: group/assignment-5/NodeTest.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class NodeTest {
+
+    @Test
+    public void testGetChildrenAsArray_NoChildren() {
```
Comment (angelina-nedelcu) [inline group/assignment-5/NodeTest.java]: This should only test Node, it should not involve another class.

File: group/assignment-5/LexiconTrie.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+     *
+     * Time complexity: O(N),
+     * space complexity: O(N),
+     * where N is a number of nodes in the trie
+     * 
+     * @param node is a node whose children a checked
+     * @param sequences is list of ordered sequences of letters(rules)
+     */
+    
+    private void findCombinationsHelper(Node node, List<String> sequences) {
```
Comment (angelina-nedelcu) [inline group/assignment-5/LexiconTrie.java]: Very nice way to find this rules!

File: group/assignment-5/LexiconTrieTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class LexiconTrieTest {
```
Comment (angelina-nedelcu) [inline group/assignment-5/LexiconTrieTest.java]: Well done adding tests!

File: group/assignment-5/LexiconTrieTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    @Test
+    public void test_EmptyTrie() {
```
Comment (angelina-nedelcu) [inline group/assignment-5/LexiconTrieTest.java]: Names in java are not test_something, but testSomething. I know it looks more readable with _ , but it is good to have consistency.

File: group/assignment-5/Node.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        char[] arr = new char[children.size()];
+        for (Map.Entry<Character, Node> child: children.entrySet()) {
+            arr[child.getValue().getNumber()] = child.getKey();
+        }
+        return convertToString(arr);
+    }
+
+    public static String convertToString(char[] arr) {
+        StringBuilder sb = new StringBuilder();
```
Comment (Utena) [inline group/assignment-5/Node.java]: I propose (more readable):

public static String convertToString(char [] arr) {

       String string = new String(arr);
       return string;
}

**Pull request closed without merging**